src/transforms/filter.jl

```diff
@@ -0,0 +1,51 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Filter(f, table)
+
+The transform that filters the rows based on a given function.
+"""
+struct Filter <: Transform
+    f::Base.Callable
+
+
+    function Filter(new_filter)
+        new(new_filter)
+    end
+
+    function Filter()
+        new(_default_filter)
+    end
```

Comment:
Please use the code style of the package. 2 whitespaces for indentation, no double white lines, etc.

Reply:
Also, please define outer constructors instead of inner constructors and make the field `f` of type F as a type parameter.

Reply:
Please rename `f` to `pred` for predicate function.

Reply:
> Also, please define outer constructors instead of inner constructors and make the field `f` of type F as a type parameter.

Are these two for the sake of consistency or is there a practical reason?